realm/realm-library/src/main/java/io/realm/Realm.java

```diff
@@ -291,6 +295,8 @@ private static Realm createAndValidate(RealmConfiguration configuration, Boolean
 
             // Increment global reference counter
             realm.acquireFileReference(configuration);
```

Comment:
The code would be cleaner if this was moved to acquireRealmFileReference i think?